sympy/functions/elementary/piecewise.py

```diff
@@ -1029,6 +1029,103 @@ def _eval_rewrite_as_ITE(self, *args, **kwargs):
             last = ITE(c, a, last)
         return _canonical(last)
 
+    def _eval_rewrite_as_KroneckerDelta(self, *args):
+        from sympy import Ne, Eq, Not, KroneckerDelta
+
+        class UnrecognizedCondition(Exception):
+            pass
+
+        def rewrite_eq(args):
+            return KroneckerDelta(*args)
+
+        def rewrite_ne(args):
+            return 1 - KroneckerDelta(*args)
+
+        def rewrite_and(args):
+            k = 1
+            for c in args:
+                if isinstance(c, Eq):
+                    k *= rewrite_eq(c.args)
+                elif isinstance(c, Ne):
+                    k *= rewrite_ne(c.args)
```

Comment:
Since this if/elif chain appears 4 times in the code I think that this code can be simplified by having a rewrite function that distributes the 4 cases to the 4 different `rewrite_*` functions

Reply:
I was just thinking the same thing.